src/sonarcloud/token.ts

```diff
@@ -0,0 +1,38 @@
+import keytar from "keytar"
+
+export interface SonarCloudTokenProvider {
+  getToken(): Promise<string | undefined>
+  markInvalid(): Promise<void>
+}
+
+export class SonarCloudTokenCliProvider implements SonarCloudTokenProvider {
+  private keyringService = "cals"
+  private keyringAccount = "sonarcloud-token"
+
+  async getToken(): Promise<string | undefined> {
+    if (process.env.CALS_SONARCLOUD_TOKEN) {
+      return process.env.CALS_SONARCLOUD_TOKEN
+    }
+
+    const result = await keytar.getPassword(
```

Comment:
If you remove the CLI commands you may want to consider removing all of the keychain specific stuff here as well.